functions.php

```diff
@@ -0,0 +1,102 @@
+<?
+
+/*
+ * Returns a new PDO connection
+ *
+ * @return PDO
+ */
+function getDb() :PDO
+{
+    $db = new PDO('mysql:host=db;dbname=booker_winners', 'root', 'password');
+    $db->setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
+    $db->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
+    return $db;
+}
+
+/* displays list of DB info
+ *
+ * @param $winners array
+ *
+ * @return string of all winners
+ */
+function display_winners(array $winners) :string
+{
+    $output = '';
+    if(empty($winners)) {
+        $output = 'There are no winners.';
+    } else {
+        foreach ($winners as $winner) {
+            $output .= '<h2>Prize year: ' . $winner['prize_year'] . '</h2>';
+            $output .= '<div><span>Author: ' . $winner['author_name'] . '</span></div>';
+            $output .= '<div><span>Title: ' . $winner['book_name'] . '</span></div>';
+            $output .= '<div><span>Author nationality: ' . $winner['author_nationality'] . '</span></div>';
+        }
+    }
+    return $output;
+}
+
+/*
+ * validates user input on server side
+ *
+ * @param string data by reference
+ *
+ * @return validated entries
+ */
+function validate_input(string $data) :string
+{
+    $data = trim($data);
+    $data = stripslashes($data);
+    $data = htmlspecialchars($data);
+    return $data;
+}
+
+/*
+ * if input received, passes each field entry through validation function
+ *
+ * @return array of validated user input
+ */
+function test_input() :array
+{
+    if (count($_POST) > 0) {
+      {
+            $_POST['prize_year'] = validate_input($_POST['prize_year']);
+            $_POST['author_name'] = validate_input($_POST['author_name']);
+            $_POST['book_name'] = validate_input($_POST['book_name']);
+            $_POST['author_nationality'] = validate_input($_POST['author_nationality']);
```

Comment:
It's generally considered to be bad practice to edit what is inside of $_POST - its contents should always be dictated by the user.

Instead, I would suggest adding them to a new array and then returning that instead.